sparse/tests/test_dok.py

```diff
@@ -88,19 +119,24 @@ def test_getitem_single(shape, density):
         assert np.isclose(s[idx], x[idx])
 
 
-@pytest.mark.parametrize(
-    "shape, density, indices",
-    [
-        ((2, 3), 0.5, (slice(1),)),
-        ((5, 5), 0.2, (slice(0, 4, 2),)),
-        ((10, 10), 0.2, (slice(5), slice(0, 10, 3))),
-        ((5, 5), 0.5, (slice(0, 4, 4), slice(0, 4, 4))),
-        ((5, 5), 0.4, (1, slice(0, 4, 1))),
-        ((10, 10), 0.8, ([0, 4, 5], [3, 2, 4])),
-        ((10, 10), 0, (slice(10), slice(10))),
-    ],
-)
-def test_getitem(shape, density, indices):
+# @pytest.mark.parametrize(
+#     "shape, density, indices",
+#     [
+#         ((2, 3), 0.5, (slice(1),)),
+#         ((5, 5), 0.2, (slice(0, 4, 2),)),
+#         ((10, 10), 0.2, (slice(5), slice(0, 10, 3))),
+#         ((5, 5), 0.5, (slice(0, 4, 4), slice(0, 4, 4))),
+#         ((5, 5), 0.4, (1, slice(0, 4, 1))),
+#         ((10, 10), 0.8, ([0, 4, 5], [3, 2, 4])),
+#         ((10, 10), 0, (slice(10), slice(10))),
+#     ],
+# )
+@given(data())
+def test_getitem(data):
+    n = st.integers(min_value=1, max_value=5)
```

Comment:
Yup this works

Reply:
This can be simplified and generalized at the same time:

```python
@composite
def dok_shape_items():
    shape = draw(array_shapes(min_dims = 1, max_dims=3, min_side=1, max_side=10))
    items = draw(dictionaries(
        keys=basic_indices(shape),
        values=floats(),
        min_size=0,
        max_size=0.5 * np.prod(shape)
    ))
    return shape, items
```

And then use it as any other strategy. This also allows it to be re-usable.